### PR TITLE
torrent transport

### PR DESCRIPTION
**Pull request by pigmej**
It's the same idea/code for torrent itself as in #117 but ported to BAT idea.

* Torrent client + libs installation on `torrent_transport.run` action.
* Torrent + BAT
* Some adjustments in BAT transport

### CODE REVIEW

**Comment by pigmej** [comment]
To test:

```
python examples/torrent/example.py
solar changes stage
solar changes process
solar orch run-once last
```

Then on `solar-dev` you should have new entry in `/etc/hosts`.

File: examples/torrent/example.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# To run:
```
Comment (loles) [inline examples/torrent/example.py]: move it to readme and add correct paths

File: examples/torrent/example.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    hosts = vr.create('hosts_file', 'resources/hosts_file', {})[0]
+
+    # let's add torrent transport for hosts file deployment (useless in real
+    # life)
+
+    torrent_transport = vr.create('torrent_transport',
+                                  'resources/transport_torrent',
+                                  {'trackers': ['http://tracker01-bud.infra.mirantis.net:8080']})[0]
+
```
Comment (loles) [inline examples/torrent/example.py]: Please add explenation here what is this tracker?

File: resources/transport_torrent/scripts/solar_torrent.py [hunk 3]
```diff
@@ -0,0 +1 @@
+../../../solar/solar/core/transports/helpers/solar_torrent.py
```
Comment (loles) [inline resources/transport_torrent/scripts/solar_torrent.py]: :o

Comment (pigmej) [inline resources/transport_torrent/scripts/solar_torrent.py]: What is wrong with this ?

Comment (loles) [inline resources/transport_torrent/scripts/solar_torrent.py]: nothing, it's cool :p

File: examples/torrent/example.py (hunk 1 above)
Comment (pigmej) [inline examples/torrent/example.py]: Done

**Comment by loles** [comment]
tested with torrent example and riak example. +1

File: solar/solar/core/transports/torrent.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+import libtorrent as lt
+import os
+from uuid import uuid4
+
+
+# TODO: settings?
+# TRACKERS = ['udp://192.168.4.140:8000',
+#             'http://192.168.4.140:8000/announce']
+
+TRACKERS = ['http://tracker01-bud.infra.mirantis.net:8080']
```
Comment (dshulyak) [inline solar/solar/core/transports/torrent.py]: how this is going to be configured?

File: solar/solar/core/transports/torrent.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+
+# def common_path(paths, sep=os.path.sep):
```
Comment (dshulyak) [inline solar/solar/core/transports/torrent.py]: why not to remove this code?

File: solar/solar/core/transports/torrent.py (hunk 4 above)
Comment (dshulyak) [inline solar/solar/core/transports/torrent.py]: hm, why do we need it, if this is configured in example?

Comment (pigmej) [inline solar/solar/core/transports/torrent.py]: Sorry, I did something wrong during rebasing or porting. Should be fine now.

File: solar/solar/core/transports/torrent.py (hunk 5 above)
Comment (pigmej) [inline solar/solar/core/transports/torrent.py]: It was a leftover from other ideas.

**Pull request merged**